NDcPP_v2_2e.adoc

```diff
@@ -2941,24 +2943,79 @@ _These requirements will be revisited as new TLS versions are standardized by th
+
+_The TOE shall not negotiate sessions using any of the SSL versions as well as TLSv1.0._
+
+*FCS_TLSS_EXT.1.2* The TSF shall authenticate itself using X.509 certificate(s) using [selection: _RSA with key size [selection: 2048 bits, 3072 bits, 4096 bits]; ECDSA over NIST curves [selection: secp256r1, secp384r1, secp521r1] and no other curves_].
 
 *_Application Note {counter:appnote_count}_*
 
+_The selected algorithms and key sizes must be consistent with FCS_COP.1/SigGen and FCS_CKM.1. NDcPP does not support raw public keys as defined in RFC8446._
+
+_If the ST includes TLS 1.1 or TLS 1.2 in FCS_TLSS_EXT.1.1, the selected algorithms must be consistent with the ciphersuites selected in FCS_TLSS_EXT.1.1._
+
+*FCS_TLSS_EXT.1.3* The TSF shall perform key exchange using: [selection: 
+
+* _RSA key establishment with key size [selection: 2048 bits, 3072 bits, 4096 bits];_ 
 
-_All SSL versions and TLS v1.0 are denied. Any TLS versions not selected in FCS_TLSS_EXT.1.1 should be selected here. (If ‘none’ is the selection for this element then the ST author may omit the words “and none”.)_
+* _EC Diffie-Hellman key agreement over NIST curves [selection: secp256r1, secp384r1, secp521r1] and no other curves;_ 
 
-*FCS_TLSS_EXT.1.3* The TSF shall perform key establishment for TLS using [selection: _RSA with key size_ [selection: _2048 bits, 3072 bits, 4096 bits_]_, Diffie-Hellman parameters with size_ [selection: _2048 bits, 3072 bits, 4096 bits, 6144 bits, 8192 bits_]_, Diffie-Hellman groups_ [selection: _ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192, no other groups_]_, ECDHE curves_ [selection: _secp256r1, secp384r1, secp521r1] and no other curves_]].
+* _Diffie-Hellman parameters [selection: of size 2048 bits, of size 3072 bits, of size 4096 bits, of size 6144 bits, of size 8192 bits, ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192]_
+
+].
 
 *_Application Note {counter:appnote_count}_*
 
-The appropriate options shall be selected in the ST according to the key establishment options supported by the TOE. FMT_SMF.1 requires the configuration of the key agreement parameters to establish the security strength of the TLS connection.
+_The selected algorithms and key sizes must be consistent with FCS_CKM.2_
+
+_The following table provides the ST author with instructions for completing the selections._
 
-*FCS_TLSS_EXT.1.4* The TSF shall support [selection: _no session resumption or session tickets, session resumption based on session IDs according to RFC 4346 (TLS1.1) or RFC 5246 (TLS1.2), session resumption based on session tickets according to RFC 5077_].
+[cols=",,,",options="header",]
+|===
+| |RSA key establishment... |EC Diffie-Hellman key agreement... |Diffie-Hellman key agreement...
+|TLS 1.3 without TLS 1.2 or TLS 1.1 | Shall not select | Shall select at least one  | 
+|TLS 1.3 with TLS 1.2 or TLS 1.1 |See the following rows |Shall select at least one- see the following rows for additional instructions |
```

Comment:
@qtrinh3 The syntax for merging cells can be found here, https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#ex-cell-format.